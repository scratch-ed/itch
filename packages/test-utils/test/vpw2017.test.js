/* Copyright (C) 2020 Ghent University - All Rights Reserved */
const { runTest: originalRunner } = require('./integration-runner.js');

function runTest(projectName = undefined, testName = undefined) {
  return originalRunner(projectName, testName, { skipTranslations: true, debug: false });
}

jest.setTimeout(50000);

test('01_mad_hatter', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 29);
    expect(result).toMatchSnapshot();
  });
});

test('03_teken_een_vierkant', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 3);
    expect(result).toMatchSnapshot();
  });
});

test('04_teken_een_driehoek', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 2);
    expect(result).toMatchSnapshot();
  });
});

test('06_voetballende_kat', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 2);
    expect(result).toMatchSnapshot();
  });
});

test('07_heksenjacht', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 4);
    expect(result).toMatchSnapshot();
  });
});

test('07_heksenjacht_eenvoudig', () => {
  return runTest('07_heksenjacht', '07_heksenjacht_eenvoudig').then((result) => {
    expect(result).exactStatuses('correct', 6);
    expect(result).toMatchSnapshot();
  });
});

test('08_op_bezoek_bij_devin', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 4);
    expect(result).toMatchSnapshot();
  });
});

test('09_flauw_mopje', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 10);
    expect(result).toMatchSnapshot();
  });
});

test('11_vang_de_appels', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 7);
    expect(result).toMatchSnapshot();
  });
});

test('13_cijfersom', () => {
  return runTest().then((result) => {
    expect(result).exactStatuses('correct', 3);
    expect(result).toMatchSnapshot();
  });
});
