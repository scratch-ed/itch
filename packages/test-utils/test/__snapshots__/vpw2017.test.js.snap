// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`01_mad_hatter 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Controle op bestaande code",
    "tags": [],
    "visibility": "summary",
  },
  {
    "command": "start-group",
    "name": "Stage",
    "sprite": "Stage",
    "tags": [],
    "visibility": "summary",
  },
  {
    "command": "start-test",
    "name": "Bestaande sprite",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top, de sprite Stage is niet verwijderd.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Voorgeprogrammeerde blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Je hebt niets veranderd aan de voorgeprogrammeerde blokjes van de sprite Stage.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Controle op bestaande code",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top! Je hebt niets veranderd aan de blokjes van sprite Stage.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Rondsligerende blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Goed zo! Je hebt geen losse blokjes laten rondslingeren.",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "start-group",
    "name": "Nori",
    "sprite": "Nori",
    "tags": [],
    "visibility": "summary",
  },
  {
    "command": "start-test",
    "name": "Bestaande sprite",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top, de sprite Nori is niet verwijderd.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Voorgeprogrammeerde blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Je hebt niets veranderd aan de voorgeprogrammeerde blokjes van de sprite Nori.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Controle op bestaande code",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top! Je hebt niets veranderd aan de blokjes van sprite Nori.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Rondsligerende blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Goed zo! Je hebt geen losse blokjes laten rondslingeren.",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "start-group",
    "name": "Hat",
    "sprite": "Hat",
    "tags": [],
    "visibility": "summary",
  },
  {
    "command": "start-test",
    "name": "Bestaande sprite",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top, de sprite Hat is niet verwijderd.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Voorgeprogrammeerde blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Je hebt niets veranderd aan de voorgeprogrammeerde blokjes van de sprite Hat.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Controle op bestaande code",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Top! Je hebt niets veranderd aan de blokjes van sprite Hat.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Rondsligerende blokjes",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Goed zo! Je hebt geen losse blokjes laten rondslingeren.",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-group",
    "summary": "De bestaande code is niet gewijzigd.",
  },
  {
    "command": "start-group",
    "name": "Testen voor de Hoed",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Hoed verandert bij klikken",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Kostuums van de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Klikken op de hoed",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte blokken",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`03_teken_een_vierkant 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.2165789609714974",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Aantal vierkanten",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Er is één vierkant getekend",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Zijdes hebben lengte 200",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De zijdes hebben lengte 200",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "De pen werd gebruikt",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De pen wordt gebruikt",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`04_teken_een_driehoek 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.2165789609714974",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Er werd minstens 1 driehoek getekend.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "De pen werd gebruikt",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De pen wordt gebruikt",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`06_voetballende_kat 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.055085085131258336",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "De kat raakt de voetbal",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Op het einde van de uitvoer raakt de kat de voetbal.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 0",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Afstand van kat tot voetbal wordt kleiner",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`07_heksenjacht 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.41008109820294864",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Test 0",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De Heks beweegt niet voor de klik",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 1",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks heeft bewogen",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "na 1 seconde",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "na 2 seconden",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "tijdens een seconde",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks mag maar eenmalig veranderen van positie elke seconde",
    "status": "wrong",
  },
  {
    "command": "start-test",
    "name": "tijdens een seconde",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks mag maar eenmalig veranderen van positie elke seconde",
    "status": "wrong",
  },
  {
    "command": "start-test",
    "name": "tijdens een seconde",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks mag maar eenmalig veranderen van positie elke seconde",
    "status": "wrong",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`07_heksenjacht_eenvoudig 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.41008109820294864",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Heks bestaat",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 0",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De Heks beweegt niet voor de klik",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 1",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks is veranderd van positie.",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 2",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De heks is veranderd van positie.",
    "status": "correct",
  },
  {
    "command": "start-group",
    "name": "Blokjes",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Test 3",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Gebruik van een lus",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Test 4",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Correcte gebruik van de lus",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`08_op_bezoek_bij_devin 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.6256459238126899",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Devin stelt zich voor",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Devin vraagt je naam",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Devin zegt hallo",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Devin stelt zich minstens 2 seconden lang voor",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "OK",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`09_flauw_mopje 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.0870063972180754",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Bericht nummer 0 komt van de Kat",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Spreker 0 is de Kat zijn!",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte text in bericht nummer 0",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De tekst in de tekstballon is juist",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Bericht nummer 1 komt van de Bananen",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Spreker 1 is de Bananen zijn!",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte text in bericht nummer 1",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De tekst in de tekstballon is juist",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Bericht nummer 2 komt van de Kat",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Spreker 2 is de Kat zijn!",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte text in bericht nummer 2",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De tekst in de tekstballon is juist",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Bericht nummer 3 komt van de Bananen",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Spreker 3 is de Bananen zijn!",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte text in bericht nummer 3",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De tekst in de tekstballon is juist",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Bericht nummer 4 komt van de Kat",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Spreker 4 is de Kat zijn!",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Correcte text in bericht nummer 4",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De tekst in de tekstballon is juist",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`11_vang_de_appels 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Andere vakjes",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Niet geprutst met andere sprites",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "De variable Score moet bestaan",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Er is een variabele met de naam Score.",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "start-group",
    "name": "Groep 0.9454757552594278",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "Score moet starten op 0",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Kom moet starten op 0, -150",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Appels komen bovenaan tevoorschijn",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "Appels komen bovenaan tevoorschijn",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Appels komen op willekeurige positie",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Eindscore",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De score eindigt op 10",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "Score stijgt",
    "tags": [],
  },
  {
    "command": "close-test",
    "feedback": "De score stijgt met één",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;

exports[`13_cijfersom 1`] = `
[
  {
    "command": "start-judgement",
    "version": 2,
  },
  {
    "command": "start-group",
    "name": "Groep 0.15106424790626252",
    "tags": [],
    "visibility": "show",
  },
  {
    "command": "start-test",
    "name": "De berekening van cijfersom 2586",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "De berekening van cijfersom 000",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "start-test",
    "name": "De berekening van cijfersom 199999",
    "tags": [],
  },
  {
    "command": "close-test",
    "status": "correct",
  },
  {
    "command": "close-group",
  },
  {
    "command": "close-judgement",
  },
]
`;
